demux/demux_lavf.c

```diff
@@ -466,9 +466,12 @@ static int lavf_check_file(demuxer_t *demuxer, enum demux_check check)
         }
     }
 
+    // HLS streams seems to be not well tagged, so matching mime type is not
```

Comment:
another nitpick: afaik the spec allows for either the mime type *or* extension to be set, so it's not accurate to put the blame on producers

Reply:
HTTP resources are identified by Content-Type. There is no notion of "extension", links doesn't map to files 1:1 almost never. Of course RFC allows mime type or extension, because for all other protocols extension matching is the way.

If provider sends `application/x-mpeg` (real example) or some other random shit I will blame them, matching extension in for HTTP resources is workaround for this and nothing else.

I made this patch because after all it is correct way to do the detection, but I fell like comment that "HLS streams seems to be not well tagged" is accurate in a sense some of the streams doesn't have Content-Type properly set. Not putting blame, but stating the fact it is not enough to use it.

Reply:
Well ok.